drivers/gpio/gpio_sx1509b.c

```diff
@@ -436,9 +430,30 @@ static int sx1509b_init(struct device *dev)
 
 	k_sleep(K_MSEC(RESET_DELAY_MS));
 
+	/* Reset state */
+	drv_data->pin_state = (struct sx1509b_pin_state) {
+		.dir = (ALL_PINS
+			& ~(DT_INST_0_SEMTECH_SX1509B_INIT_OUT_LOW
+			    | DT_INST_0_SEMTECH_SX1509B_INIT_OUT_HIGH)),
+		.data = (ALL_PINS
+			 & ~DT_INST_0_SEMTECH_SX1509B_INIT_OUT_LOW),
```

Comment:
Perhaps it would be good to add a build assertion checking if the outputs to be initially driven low and those to be driven high do not overlap?

Reply:
Do you want to insist on this?  I don't think it's worthwhile: the resulting configuration is not unstable (pins in both will be set low), and presumably the problem will be discovered during testing.

All other comments have been addressed.

Reply:
> Do you want to insist on this?

No, it was just a suggestion.